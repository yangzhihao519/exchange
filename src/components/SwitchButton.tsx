import React from 'react'

interface SwitchButtonProps {
    onSwitchExchange: () => void
}

class SwitchButton extends React.Component<SwitchButtonProps>{

    render() {
        return (<div className='switch-button' onClick={this.props.onSwitchExchange} />)
    }
}

export default SwitchButton