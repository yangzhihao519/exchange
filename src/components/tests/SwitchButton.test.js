import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import SwitchButton from '../SwitchButton';

const mockFunction = jest.fn();

describe('SwitchButton', () => {
  it('renders without crashing given the required props', () => {
    const props = {
        onSwitchExchange: mockFunction
    }
    const wrapper = shallow(<SwitchButton {...props} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})