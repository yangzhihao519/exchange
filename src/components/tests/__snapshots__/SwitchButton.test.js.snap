// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SwitchButton renders without crashing given the required props 1`] = `
<div
  className="switch-button"
  onClick={[MockFunction]}
/>
`;
